### PR TITLE
New feature #12560: enable video in spite of active xss filtering

### PR DESCRIPTION
**Pull request by gabrieljenik**
Added new configuration to the HtmlPurifier.
Extended the standard HtmlPurifier so it exposes the config in a public method.

### CODE REVIEW

File: application/core/LSYii_HtmlPurifier.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ * License: GNU/GPL License v2 or later, see LICENSE.php
+ * LimeSurvey is free software. This version may have been modified pursuant
+ * to the GNU General Public License, and as distributed it includes or
+ * is derivative of works licensed under the GNU General Public License or
+ * other free or open source software licenses.
+ * See COPYRIGHT.php for copyright notices and details.
+ *
+ */
+
+class LSYii_HtmlPurifier extends CHtmlPurifier
```
Comment (Shnoulle) [inline application/core/LSYii_HtmlPurifier.php:17]: :+1:

File: application/core/LSYii_Validators.php [hunk 2]
```diff
@@ -128,6 +128,44 @@ public function xssFilter($value)
         );
         // To allow script BUT purify : HTML.Trusted=true (plugin idea for admin or without XSS filtering ?)
 
+        // Enable video
```
Comment (Shnoulle) [inline application/core/LSYii_Validators.php:131]: Just a question : can not be in `LSYii_HtmlPurifier` ?

**Review by gabrieljenik — COMMENTED** [review]
What about now?

**Comment by Shnoulle** [comment]
I think only for https://github.com/LimeSurvey/LimeSurvey/pull/1589/files#diff-a832a00a927d6b1c5bb94e437eb7881fR54 but : it's just a question :)

**Comment by gabrieljenik** [comment]
Sorry, what you mean? Don't understand.

**Comment by Shnoulle** [comment]
Something like a enableVideo function

https://github.com/LimeSurvey/LimeSurvey/pull/1589/files#diff-a832a00a927d6b1c5bb94e437eb7881fR54-R90

```
public function enableVideo() {
    $config = $this->getConfig();
    // Do the job
}
```

**Comment by gabrieljenik** [comment]
It is a possibility. Stil, I believe xss filtering should have almost always the same filters. That's why I didn't include that method.
So, this new class is more like a factory, which gives you filters based on scenarios. We could have it return based on individual settings, as for example if video is allowed or not, but I think that could be left for later, right?

**Comment by Shnoulle** [comment]
Oh yes, sure !

I just ask a question, not make an advice ...

**Pull request merged**